docs/examples/vectorfields.py

```diff
@@ -0,0 +1,493 @@
+
+        k0Switch = (z + 1) % 2 * (zp + 1) % 2
+        k1Switch = z * zp
+
+        return k0Switch * self.kernel0(X, Xp) + k1Switch * self.kernel1(X, Xp)
+
+
+# %% [markdown]
+# ### GPJax Implementation
+# Next, we define the model in GPJax. The prior is defined using $k_{\text{vel}}\left(\mathbf{X}, \mathbf{X}^\prime \right)$ and 0 mean. We choose a Gaussian likelihood.
+#
+
+# %%
+# Define the Helmholtz GP
+kernel = VelocityKernel()
+mean = gpx.mean_functions.Zero()
+Prior = gpx.Prior(mean_function=mean, kernel=kernel)
+Likelihood = gpx.Gaussian(
+    num_datapoints=D0.n, obs_noise=jnp.array([1.0], dtype=jnp.float64)
+)
+Posterior = Prior * Likelihood
+
+
+# %% [markdown]
+# With a model now defined, we can proceed to optimise the hyperparameters of our likelihood over $D_0$. This is done by minimising the marginal log likelihood using `optax`. We also plot its value at each step to visually confirm that we have found the minimum
+
+# %%
+
+# define the Marginal Log likelihood using D0
+Objective = gpx.objectives.ConjugateMLL(negative=True)
+Objective(Posterior, train_data=D0)
+Optimiser = ox.adam(learning_rate=0.01)
+Objective = jit(Objective)
+
+NIters = 10000
+# Optimise to minimise the MLL
+OptPosteriorVel, history = gpx.fit(
+    model=Posterior,
+    objective=Objective,
+    train_data=D0,
+    optim=Optimiser,
+    num_iters=NIters,
+    safe=True,
+    key=key,
+)
+
+fig, ax = plt.subplots(1, 1)
+ax.plot(history, color="red")
+ax.set(xlabel="Training iteration", ylabel="Negative marginal log likelihood")
+plt.show()
+
+# %% [markdown]
+# ### Comparison
+# We next obtain the latent distribution of the GP of $g$ at $\mathbf{x}_{0,i}$, then extract its mean and standard at the test locations, $\mathbf{F}_{\text{latent}}(\mathbf{x}_{0,i})$, as well as the standard deviation (we will use it at the very end).
+
+# %%
+VelocityLatent = OptPosteriorVel.predict(PosTest3D, train_data=D0)
+VelocityMean = VelocityLatent.mean()
+VelocityStd = VelocityLatent.stddev()
+# extract x0 and x1 values over g
+Vel_Lat = [VelocityMean[::2].reshape(17, 17), VelocityMean[1::2].reshape(17, 17)]
+Pos_Lat = Pos_Test
+
+# %% [markdown]
+# We now replot the ground truth (testing data) $D_0$, the predicted latent vector field $\mathbf{F}_{\text{latent}}(\mathbf{x_i})$, and a heatmap of the residuals at each location $R(\mathbf{x}_i) = \left|\left| \mathbf{y}_{0,i} - \mathbf{F}_{\text{latent}}(\mathbf{x}_i) \right|\right|$.
+
+# %%
+
+# Residuals between ground truth and estimate
+ResidualsVel = jnp.sqrt(
+    (Vel_Test[0].reshape(17, 17) - Vel_Lat[0]) ** 2
+    + (Vel_Test[1].reshape(17, 17) - Vel_Lat[1]) ** 2
+)
+X = Pos_Test[0]
+Y = Pos_Test[1]
+
+# make figure
+fig, ax = plt.subplots(1, 3)
+fig.tight_layout()
+# ground truth
+ax[0].quiver(
+    Pos_Train[0],
+    Pos_Train[1],
+    Vel_Train[0],
+    Vel_Train[1],
+    color="red",
+    label="Testing Data",
+)
+ax[0].quiver(Pos_Test[0], Pos_Test[1], Vel_Test[0], Vel_Test[1], label="Training data")
```

Comment:
Using the `plot_vector_field` helper function as described in https://github.com/JaxGaussianProcesses/GPJax/pull/341/files#r1284415145 would simplify this as
```suggestion
plot_vector_field(ax[0], data_train, color=colors[1], label="Testing Data")
plot_vector_field(ax[0], data_test, color=colors[0], label="Training data")
```
(better to be explicit about the choice of color)